calculator/interfaces/CalculatorDateAndTime.java

```diff
@@ -0,0 +1,125 @@
+package calculator.interfaces;
+import java.time.ZonedDateTime;
+import java.time.ZoneId;
+import java.util.regex.*;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAdjusters;
+import java.time.DayOfWeek;
+import java.time.Duration;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.LocalDate;
+public class CalculatorDateAndTime implements Calculator {
+
+	private final Pattern pattern = Pattern.compile("^([\\+\\-])\\s*((\\d*)y)?\\s*((\\d*)m)?\\s*((\\d*)d)?\\s*((\\d*)h)?\\s*((\\d*)min)?\\s*((\\d*)s)?$");
+	
+	private ZonedDateTime zone = ZonedDateTime.now();
+
+	public CalculatorDateAndTime (Zone zone) {
+		this.zone = ZonedDateTime.now(ZoneId.of(zone.idZone));
+	}
+	public String toString () {
+		return zone.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
+	}
+    public String whenSunday () {
+        return "Sunday is " + zone.with(TemporalAdjusters.next(DayOfWeek.SUNDAY)).format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+    }
+    public void showSecondsSince (String input) {
+        Pattern yearMonthDay = Pattern.compile("^showSecondsSince(\\d{1,4})\\s?([1-9]|1[0-2])?\\s?([1-9]|1[0-9]|2[0-9]|3[01])?$");
```

Comment:
please describe what is that regex accepting and add few examples of correct input. (In code as comment, do that for every long more complicated regex)

Reply:
and compile pattern once, it is rather, compiling pattern is rather slow, im not sure though :)

Reply:
rename pattern to better describe what it is, maybe showSecondsSincePattern ?

Reply:
put every pattern string to static final variable
static final SHOW_SECONDS_SINCE_PATTERN_STR = "^showSecondsSince(\\d{1,4})\\s?([1-9]|1[0-2])?\\s?([1-9]|1[0-9]|2[0-9]|3[01])?$";
that pattern is also used in main method, reuse that variable there